L1Trigger/TrackFindingTracklet/python/l1tTTTracksFromTrackletEmulation_cfi.py

```diff
@@ -21,7 +21,7 @@
                                                processingModulesFile = cms.FileInPath('L1Trigger/TrackFindingTracklet/data/processingmodules_hourglassExtended.dat'),
                                                wiresFile = cms.FileInPath('L1Trigger/TrackFindingTracklet/data/wires_hourglassExtended.dat'),
                                                # Quality Flag and Quality params
-                                               TrackQuality = cms.bool(True),
+                                               TrackQuality = cms.bool(False),
```

Comment:
did you mean to turn this off by default?

Reply:
Yes, this turns off the simulation TQ from [running in L1FPGATrackProducer.cc](https://github.com/cms-L1TK/cmssw/blob/L1TK-dev-12_6_0_pre5/L1Trigger/TrackFindingTracklet/plugins/L1FPGATrackProducer.cc#L323) which does not have full track information (chi2 variables calculated in KFout with newKF for example). Instead, the track quality now runs after the KFout module when the TTTracks are created from the KFTracks. There is no bool to turn this track quality on or off so it runs by default when newKF is selected.

Although I now realize that this will turn of track quality when running the old KF which is not necessarily what we want. Perhaps instead of changing it here, we should manually set it to False in the L1NtupleMaker_cfg when newKF is selected?

Reply:
Like you mentioned, I'd suggest to keep the TrackQuality bool **On** by default and modify the cfg.py to turn it off when new KF is used. There is an example here:
https://github.com/cms-L1TK/cmssw/blob/cf9087e16493a7b4002cce06537f99d3a9c79912/L1Trigger/TrackFindingTracklet/test/L1TrackNtupleMaker_cfg.py#L190

Reply:
@cgsavard I suspect this PR may be based on a misunderstanding. 
When using HYBRID_NEWKF, the Kalman Filter consists of three EDProducers run in succession ProducerKFin, ProducerKF & ProducerKFout. All the KF fit maths are in ProducerKF. ProducerKFin & ProducerKFout do data formatting. However, ProducerKFout also calls the TrackQuality code at https://github.com/cms-L1TK/cmssw/blob/L1TK-dev-12_6_0_pre5/L1Trigger/TrackFindingTracklet/plugins/ProducerKFout.cc#L299 .

The KFin, KF & KFout each write an EDProduct consisting of streams of 64 bit data words, containing the digitised fitted track data. After the KF is run and before KFout is run, an EDProducer called ProducerTT is run that converts the 64b EDProduct written by the KF into a TTTrack collection.

To fix the problem that the TTTrack collection doesn't contain TrackQuality info, the ProducerTT (and probably the ProducerAS that is run just after it) should be called after KFout instead of after KF. They may need rewriting slightly to cope with this change. @tschuh is the best person to advise on this.

The solution proposed in this PR is rather a bodge. The DataFormat class is intended for data-formatting, not for implementing track-finding functionality like running track quality. In addition, without moving ProducerTT to after ProducerKFout, any efficiency losses introduced by KFout (e.g. due to truncation) will not be reflected in the output TTTrack collection.

We have not found a volunteer yet to implement this change.

Reply:
> Like you mentioned, I'd suggest to keep the TrackQuality bool **On** by default and modify the cfg.py to turn it off when new KF is used. There is an example here:
> 
> https://github.com/cms-L1TK/cmssw/blob/cf9087e16493a7b4002cce06537f99d3a9c79912/L1Trigger/TrackFindingTracklet/test/L1TrackNtupleMaker_cfg.py#L190

@Jingyan95 Done